menpofit/io.py

```diff
@@ -0,0 +1,336 @@
+    r"""
+    Returns the remote URL for a fitter with identifier ``name``.
+
+    Parameters
+    ----------
+    name : `str`
+        The identifier of the fitter, e.g. `balanced_frontal_face_aam`
+
+    Returns
+    -------
+    url : `str`
+        The URL that this fitter is stored at for this installation of
+        menpofit
+    """
+    return '{}/menpofit/{}'.format(MENPO_URL, filename_for_fitter(name))
+
+
+def path_of_fitter(name):
+    r"""
+    Returns a path to where a fitter with identifier ``name`` can be
+    located on disk.
+
+    Parameters
+    ----------
+    name : `str`
+        The identifier of the fitter, e.g. `balanced_frontal_face_aam`
+
+    Returns
+    -------
+    path :  `Path`
+        A path to this fitter on disk
+    """
+    return menpofit_data_dir_path() / filename_for_fitter(name)
+
+
+def copy_and_yield(fsrc, fdst, length=1024*1024):
+    """copy data from file-like object fsrc to file-like object fdst"""
+    while 1:
+        buf = fsrc.read(length)
+        if not buf:
+            break
+        fdst.write(buf)
+        yield
+
+
+def download_file(url, destination, verbose=False):
+    r"""
+    Download a file from a URL to a path, optionally reporting the progress
+
+    Parameters
+    ----------
+    url : `str`
+        The URL of a remote resource that should be downloaded
+    destination : `Path`
+        The path on disk that the file will be downloaded to
+    verbose : `bool`, optional
+        If ``True``, report the progress of the download dynamically.
+    """
+    from menpo.visualize.textutils import print_progress, bytes_str
+    req = urlopen(url)
+    chunk_size_bytes = 512 * 1024
+
+    with open(str(destination), 'wb') as fp:
+
+        # Retrieve a generator that we can keep yielding from to download the
+        # file in chunks.
+        copy_progress = copy_and_yield(req, fp, length=chunk_size_bytes)
+
+        if verbose:
+            # wrap the download object with print progress to log the status
+            n_bytes = int(req.headers['content-length'])
+            n_items = int(ceil((1.0 * n_bytes) / chunk_size_bytes))
+            prefix = 'Downloading {}'.format(bytes_str(n_bytes))
+            copy_progress = print_progress(copy_progress, n_items=n_items,
+                                           show_count=False, prefix=prefix)
+
+        for _ in copy_progress:
+            pass
+
+    req.close()
+
+
+def load_fitter(name):
+    r"""
+    Load a fitter with identifier ``name``, pulling it from a remote URL if
+    needed.
+
+    Parameters
+    ----------
+    name : `str`
+        The identifier of the fitter, e.g. `balanced_frontal_face_aam`
+
+    Returns
+    -------
+    fitter : `Fitter`
+        A pre-trained menpofit `Fitter` that is ready to use.
+    """
+    path = path_of_fitter(name)
+    if not path.exists():
+        print('Downloading {} fitter'.format(name))
+        url = url_of_fitter(name)
+        print(url, path)
```

Comment:
@patricksnape it's definitely more of a debug statement that I forgot that I forgot to remove, apologies.
Not the end of the world being there, feel free to spice it up or remove it if you want to!